2-mandatory/2-financial-times.js

```diff
@@ -5,33 +5,68 @@
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
 function potentialHeadlines(allArticleTitles) {
-    // TODO
+    headlines = []; // we declare an empty array so when we get the headlines with length<=65 we push them here
+
+    for (article of allArticleTitles) { // we loop through all articles  and check their length
+        if (article.length <= 65) {
+            headlines.push(article); //for every headline that  passes the condition we push it to the empty array
+        }
+
+    }
+    return headlines; //then we return the array with the headlines that have <=65 words
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
-    (you can assume words will always be seperated by a space)
+    (you can assume words will always be separated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+    fewestWordsTitle = ''; //we use an empty string since we haven't had yet any article to display 
+    titleLengthNr = Infinity; //we can use any limit we want too but we want to make sure that the limit for the comparison it's not very small so it can fit all headlines available
+
+    headlines = potentialHeadlines(allArticleTitles) //we get the function above by saying that headlines is inside this function as mentioned
+    //console.log(headlines) i used console.log to see if my headlines was displaying or not
+
+    for (headline of headlines) { //we loop through each headline from headlines array
+        // console.log(headline)
+        titleLength = headline.split(' ').length; //we calculate te length by splitting each headline to words and  not counting spaces .
+        //console.log(titleLength)
+
+        if (titleLength < titleLengthNr) { //and compare it with the value we first gave the variable
+            fewestWordsTitle = headline; //the title with the smallest amount of words is going to be displayed in the variable fewestWords
+            titleLengthNr = titleLength; /*since we didn't have a starting value we used infinity to compare the length but after we passed the length of 
+            at least one headline we compare it then to that and then let the function know that are the same  */
+        }
+    }
+    return fewestWordsTitle; //function is going to  return the headline with the fewest words 
 }
 
 /*
-    The editor of the FT has realised that headlines which have numbers in them get more clicks!
+    The editor of the FT has realized that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    function withNumber(title) {
```

Comment:
I like that you've put the `withNumber` logic into its own function 👍 
It might be worth defining the `withNumber` function outside of the `headlinesWithNumbers` function - so it can potentially be re-used. But not a big deal!